### ISSUE TITLE
Variant value?

### ISSUE THREAD

**Issue by MzTechnology97** [NebraLtd/helium-controllino#1]
What value should I enter the variant?
I tried COMP-CONTROLLINO but it doesn't work

**Comment by shawaj** [comment]
Yes this doesn't work just yet. It's a work in progress

**Comment by uros76** [comment]
Hi. Is there a variant variable known now?

**Comment by c0r391** [comment]
would be great when this works now, because of the crappy service fee of contollino :-X

**Comment by uros76** [comment]
I am afraid this is a dead end too :( no response in months. Perhaps it's due to a lack of agreement between nebra and controllino? I don't know...

**Comment by shawaj** [comment]
Just FYI this repo is not really monitored.

But in any case, we haven't been able to get this working yet. As you can see here:

https://github.com/NebraLtd/hm-pyhelper/issues/127

**Comment by uros76** [comment]
> 

Thanks. Even that one is stuck since July :( 

Do you guys need any help with testing or anything else to get it working? There are few of us controllino users who can assist in any way we can.

**Comment by shawaj** [comment]
@uros76 yes unfortunately we haven't had the time to dedicate to another manufacturers software.

The main issue is that they use a concentrator with a power enable, which we don't currently support in our software. And we haven't actually figured out how the enable works yet.

We were hoping to get some info on this directly from controllino but didn't hear back.

**Comment by uros76** [comment]
Ah ok I understand @shawaj. Controllino was dissolved recently and the new iteration of a maker company is doing crazy stuff like making users wait months for support. Or the new crazy thing they decided to charge for support from next year. 
You can forget about them coming back to you with any technical development or clarification. In fact they would probably want to charge you for any information. It's that bad. 

That's why controllino users community is trying to find a solution to untether from this maker and their crazy ideas completely. 
I am not a tech expert sadly to help on the concentrator side. The power enable is an unique feature of Embit concentrator? 
Did you have a look at these docs? 
http://www.embit.eu/wp-content/uploads/Datasheet-EMB-LR1301-mPCIe_.pdf
http://www.embit.eu/wp-content/uploads/Datasheet-EMB-RasPI-130x-Cape.pdf

**Comment by shawaj** [comment]
@uros76 Who is the new company that took over controllino?

The datasheets are helpful thanks. They show the power enable pin and the i2c bus :-)

I think the other issue as well was that they put the ECC chip on the concentrator itself. But not sure if that will be a blocker.

I'll have to find my controllino hotspot now to check it out 😂

I can't give any idea of how long this will take though as it requires building in new functionality that we don't currently have

**Comment by shawaj** [comment]
PDFs for future reference
[Datasheet-EMB-LR1301-mPCIe_.pdf](https://github.com/NebraLtd/helium-controllino/files/10178661/Datasheet-EMB-LR1301-mPCIe_.pdf)
[Datasheet-EMB-RasPI-130x-Cape.pdf](https://github.com/NebraLtd/helium-controllino/files/10178662/Datasheet-EMB-RasPI-130x-Cape.pdf)

**Comment by uros76** [comment]
Controllino was not sold but kind of changed name with the same owner. New name is Conelcon. Good luck getting anything useful from them. Many of us have escalated issues with Helium already... 

Some users reported that they could swap rpi board and the hotspot resumed working. Not sure if that changes anything but yeah chip seems to be on the concentrator. 

Pls dig out that old controllino of yours and play around. If you need ant tips or help or some guinea pig for testing send me a DM ;)

**Comment by uros76** [comment]
> 

Of course, I have two of them. Shoot me DM and I will set up remote access for you as needed.

**Comment by uros76** [comment]
One more thing @shawaj I created an unofficial controllino community discord server back in September where I shared tons of findings and our knowledge. Check it out perhaps something useful for your discovery 
https://discord.io/controllino

**Comment by shawaj** [comment]
Can you flash the image shown here to the SD card?

https://hub.balena.io/nebraltd/helium-controllino

Click get started, then choose pi 4 as the device type

**Comment by uros76** [comment]
That link takes me to generic page. 
You mean this one? https://hub.balena.io/organizations/nebraltd/fleets/HELIUM-CONTROLLINO

**Comment by shawaj** [comment]
Yes sorry that's the correct link

**Comment by uros76** [comment]
Downloading slowly don't know why. Once I load it will report back

**Comment by shawaj** [comment]
Awesome thanks @uros76 then I'll have a poke around 🙂

**Comment by uros76** [comment]
I tried downloading the image twice and it fails on transfer. Now trying the flashing straight from the browser and etcher app. Let's see if it finally works

**Comment by shawaj** [comment]
If not try downloading from here, I just downloaded and uploaded to Google drive.

This one will only work on ethernet though...

https://drive.google.com/file/d/1fpkCHyoVbMrWjaXeyA6HbCFUHwAmtaFE/view?usp=drivesdk

**Comment by uros76** [comment]
The file from gdrive worked. I just flashed on an sd card and plugged it into my controllino. 
Do you see it online?

**Comment by uros76** [comment]
I do see a new dot on the fleet map, it must be working then :) 
Let me know what you find out. 
![image](https://user-images.githubusercontent.com/25409772/206314492-82da2338-cdb0-4e9d-8e0f-245528dcb86b.png)

**Comment by shawaj** [comment]
Thanks! I'll try and take a look tomorrow 🙂

**Comment by uros76** [comment]
@shawaj any progress? 😊

**Comment by shawaj** [comment]
@uros76 I didn't have any time to take a look yet, but I'm home alone tonight so planning to take a look in a bit. Will let you know if I make any progress :-)

**Comment by shawaj** [comment]
@uros76 good news - i have the concentrator on your unit working. it requires us to turn on the power using the power enable chip

we are going to have to add some new logic to the pyhelper module or somewhere else in order to automatically handle this based on the variant.

this may take a bit of time but will see if we can push it ahead :-)

**Comment by shawaj** [comment]
@kashifpk @KevinWassermann94 - do you have any thoughts on where this logic should go?

it is a power enable chip to turn on the power to the HAT on the controllino unit. I am thinking maybe in hm-pyhelper?

Pull in the pin number for a power enable param in hardware definition (false if not there, number of pin if there?) and then make output and set to 1?

What do you think?

**Comment by shawaj** [comment]
Yes will keep you updated. Thanks for agreeing to be the guinea pig here 👌😋

**Comment by shawaj** [comment]
@uros76 I think units should work for controllino now on this software. Will monitor your one to check but it looks like it's working to me.

It would be useful if you know anyone else with controllino units who could flash this software so we have some more to test with.

**Comment by shawaj** [comment]
@MzTechnology97 @c0r391 this should be working now just FYI

You can install by going to https://hub.balena.io/organizations/nebraltd/fleets/HELIUM-CONTROLLINO and clicking on "Get Started"

**Comment by uros76** [comment]
@shawaj AMAZING it does work. It just had activity. Let me spread the message amongst controllino community and find more testers. 
Can we fork as a private fleet and use balena dashboard? Or we need to wait for that option a bit more?

**Comment by gery800** [comment]
Hi

Have the firmware installed thanks to Uros76.
On the dashboard about the Ip of the Controllino I see Animal Name Unavaible and Errors Found is this normal? or do I have to do something else after downloading the firmware?
Radio Operation is Flase and ECC Detected also Flase

**Comment by uros76** [comment]
Where are you looking at those errors? On balena dashboard?

**Comment by gery800** [comment]
No Ip-Adress from Miner 
![Screenshot_2022-12-10-11-58-56-590_com android chrome](https://user-images.githubusercontent.com/120245125/206851729-7adb2091-4a02-474b-97dc-b071862f9934.jpg)

**Comment by uros76** [comment]
Ah no need to looks at that. I have not even opened my hotspots IP,  will probably see same. As long as it's working I don't care about local IP dashboard. I do care about balena dashboard if it will work 😃

**Comment by shawaj** [comment]
> Can we fork as a private fleet and use balena dashboard? Or we need to wait a bit longer for that option?

@uros76 yes you can fork the fleet and run your own Balena fleet and use the dashboard but you will lose the automatic updates (or have to set them up yourself using GitHub actions or whatever)

@gery800 can you share the model number shown on your hotspot and a few pictures? Also have you already onboarded this hotspot and asserted location or not?

**Comment by gery800** [comment]
The hotspot was online with the Controllino firmware and earned HNT.
Location is set.

**Comment by shawaj** [comment]
> The hotspot was online with the Controllino firmware and earned HNT. Location is set.

@gery800 Can you share pictures of the unit? And the model number?

I am taking a look at it now from the backend. I think it's possibly got corrupted while downloading the image as the miner container seems to be crashing out. Leave it powered on as I'm trying a couple of things

**Comment by uros76** [comment]
Thanks again @shawaj I tried to download the base firmware again today and it kept failing during the download process. Perhaps that's issue @gery800 is having. 

One more thing regarding forking. Do I need to set some controllino specific variable value on my fleet?

**Comment by uros76** [comment]
Never mind, found it on hardware_definitions.py 
Should be "COMP-CONTROLLINO"

**Comment by shawaj** [comment]
You will need to set:

In device variables:

VARIANT    COMP-CONTROLLINO

In device configuration:

Define DT parameters for the default overlay. Set to `"i2c_arm=on","spi=on","audio=on"`

Define DT overlays. Set to `"spi0-1cs","gpio-poweroff,gpiopin=17,active_low"`

**Comment by uros76** [comment]
Okey, thanks. Will give it a go now. 
I am downloading forked firmware and it's still alarmingly slow. I don't know is it my computer hating balena source these days...

**Comment by shawaj** [comment]
@gery800 can you also share the animal name of your unit?

There appears to be an issue with the ECC key failing some tests. It's probably just that it's configured slightly differently to how we do ours but will need to dig a little deeper.

**Comment by gery800** [comment]
> > The hotspot was online with the Controllino firmware and earned HNT. Location is set.
> 
> @gery800 Can you share pictures of the unit? And the model number?
> 
> I am taking a look at it now from the backend. I think it's possibly got corrupted while downloading the image as the miner container seems to be crashing out. Leave it powered on as I'm trying a couple of things

Ethernet Mac DC:A6:32:EF:ED:A6
Hardware Serial Number: 1000000046cb57b5

**Comment by uros76** [comment]
The firmware download keeps failing for me regardless of the device I am downloading with. That's for base firmware and my forked fleet image :( 
![image](https://user-images.githubusercontent.com/25409772/206860901-11db8893-cfea-418d-9eb5-6277353c41f3.png)
Any idea why this is happening?

**Comment by uros76** [comment]
Update on downloading problems: I connected with VPN to different country with one of my devices and now it worked fine. 
There must be something blocking balena traffic to Poland where I live

**Comment by uros76** [comment]
Something doesnt seem to be working on forked fleet hotspot: 
![image](https://user-images.githubusercontent.com/25409772/206862510-9876c503-ecb5-4873-8719-0468527d6dbe.png)

I did setup variable and device configurations: 
![image](https://user-images.githubusercontent.com/25409772/206862524-88c82cd1-87ec-4ba1-a308-7f4155902d6e.png)
![image](https://user-images.githubusercontent.com/25409772/206862550-3d309150-024a-433b-a944-02071cc87a8d.png)

@shawaj any idea what could be wrong?

**Comment by shawaj** [comment]
Looks to me like it's running older containers might need an update

**Comment by shawaj** [comment]
@uros76 just pushed an update. Try pushing the updated docker-compose https://github.com/NebraLtd/helium-controllino/blob/master/docker-compose.yml

**Comment by uros76** [comment]
I tried going back to the base firmware, not forked version. And now I get similar errors like gery above
![image](https://user-images.githubusercontent.com/25409772/206865870-1746b535-861d-4e5b-ac87-a26f73bf0bdd.png)

I will retry my forked version and push the update.

**Comment by uros76** [comment]
I reloaded my forked firmware, updated it, and still something seems to be missing or wrong:
![image](https://user-images.githubusercontent.com/25409772/206866885-692c565f-f5ff-4d58-8438-4cefffc31ceb.png)
![image](https://user-images.githubusercontent.com/25409772/206867131-4109492f-4055-4ca4-ba87-954084eaac33.png)
Device variable and configuration are set as instructed above...

**Comment by shawaj** [comment]
@uros76 yes I think gpio-poweroff isn't working quite as expected. It seems the power enable isn't working right.

I'm investigating further

**Comment by shawaj** [comment]
@uros76 try with the dtoverlay bit changed to:

`"spi0-1cs","gpio-shutdown,gpio_pin=17,active_low=1,gpio_pull=up"`

This seems to work for me

**Comment by shawaj** [comment]
@gery800 yours is working now too it looks like. Will keep an eye on it until it witnesses and beacons

**Comment by uros76** [comment]
> 

That seems to do the the trick. Hotspot dashboard has no errors now and balena dashboard log looks cleaner now. Will keep monitoring the activity.

**Comment by shawaj** [comment]
@uros76 great stuff! Fingers crossed 🤞

**Comment by gery800** [comment]
thanks Dashboard no longer shows any errors I will of course monitor and give feedback to the miner great work

**Comment by shawaj** [comment]
@gery800 great. Just need it to beacon and witness now 👌

**Comment by shawaj** [comment]
New URL is https://nebra.io/controllino FYI

**Comment by uros76** [comment]
My hotspot witnessed and beaconed. I consider this a big win, miner works fine with forked firmware. 
Massive kudos and thanks @shawaj for helping out controllino community break free from dirty makers practices.

**Comment by shawaj** [comment]
No problem and glad it's working 😊

I'm going to leave this issue open for now as I want to add some more documentation etc to this repo and this can serve as a reminder.

We hope to apply for a Helium Foundation grant when they open again to develop and maintain a fully open source firmware that supports all existing and future miners so that people always have choice.

**Comment by shawaj** [comment]
Looks like yours has beaconed and witnessed too @gery800 so all seems to be working well 👌

**Comment by shawaj** [comment]
By the way, if anyone has found this useful please consider donating some funds at https://github.com/sponsors/NebraLtd

This will help us to dedicate resources to maintaining the 3rd party helium firmwares going forward

**Comment by uros76** [comment]
@shawaj welcome. 
I have spread the news amongst controllino community on how to use nebra firmware. I believe more and more users will get on it in the following weeks

**Comment by garfield1981** [comment]
Hello,
unfortunately it doesn't work for me from my fleet. I double checked everything. It always says: "packet-forwarder KeyError: 'COMP-CONTROLLINO'"
And I can't access the dashboard over my internal network. I tried going back to the base firmware and that works with no problems. What else can I try?
![Bild1](https://user-images.githubusercontent.com/76958512/207420071-50c2922e-10bb-47ee-926d-a904de166a0f.jpg)

![bild2](https://user-images.githubusercontent.com/76958512/207420116-ed9c57cb-2d19-4889-a8ce-507dfaf2a465.jpg)
![Bildschirmfoto zu 2022-12-13 19-43-56](https://user-images.githubusercontent.com/76958512/207420168-a685e213-891c-4843-b07f-1b746b347ae0.png)
![Bildschirmfoto zu 2022-12-13 19-44-34](https://user-images.githubusercontent.com/76958512/207420209-e8730f61-0acf-40c9-9a28-936b973e6f85.png)

**Comment by shawaj** [comment]
@garfield1981 my guess would be that you are using an old release. As that error means the hm-pyhelper module is missing the controllino definition. Only an older release would be missing that.

**Comment by garfield1981** [comment]
@shawaj 
Thanks for your answer. Sorry I'm not a pro. 
I followed Uros76 instructions exactly which he published in the Controllino Discord.
Where can I see the version?
My Controllino V1 is from the end of May 22, so it is a very late V1. is that maybe a problem?

**Comment by shawaj** [comment]
@garfield1981 I haven't seen the instructions I'm afraid so can't comment on them.

But is there a reason you need to use your own Balena fleet? If you aren't very technical then I would suggest to just use the standard install by going to https://nebra.io/controllino and clicking on Get Started.

The instructions are the same as this guide for RAK, just substituting the rak link for the controllino one https://support.nebra.com/support/solutions/articles/24000078640-getting-started-with-rak-mntd

**Comment by garfield1981** [comment]
@shawaj  I'm always very interested in testing and trying out new things. 
Yes, the standard install by going to https://nebra.io/controllino also worked immediately. I've installed it again and will leave it that way for now. 
The most important thing for me is not to be dependent on Controllino anymore.
Thank you for your time and have a nice evening.

**Comment by shawaj** [comment]
@garfield1981 the Balena fleet of your own will work, but it won't auto update. You will have to push the new release every time there is an update.

So unless you want to do the manual updates and you have a specific need for having your own fleet then I'd say to just use the standard fleet we provide.

But of course, it's up to you 😉

**Comment by garfield1981** [comment]
Hello, @uros76  I followed your instructions exactly. That doesn't work for me. I tried it 5 times. 
I started from scratch and flashed the SD card again.
The image from https://hub.balena.io/organizations/nebraltd/fleets/HELIUM-CONTROLLINO works for me immediately.

**Comment by uros76** [comment]
I don't know what else could be wrong. Those are the steps I did and it worked. Are you still getting "packet-forwarder KeyError: 'COMP-CONTROLLINO'" issue?

**Comment by uros76** [comment]
Another idea. @shawaj. If someone with a private fleet enables the "Support access" do you have access to that device than and can look at it? 
I just worry that it might not be isolated case and more like above issue could pop up.

**Comment by shawaj** [comment]
@uros76 no that would be Balena support not Nebra

**Comment by uros76** [comment]
Ah ok, never mind then. Let's see if more users support same issue

**Comment by shawaj** [comment]
I'm thinking it was due to me updating this repo (splitting it into testnet and master branches). That was around the same time.

Possibly just deleting the fleet and starting again might solve it.

Also @uros76 just to check you are aware that this will require you to manually update every time we push an update?

**Comment by uros76** [comment]
Yes fully aware, thanks. I am using nebra rak/mntd firmware on my private fleet for months now.

**Comment by shawaj** [comment]
> Yes fully aware, thanks. I am using nebra rak/mntd firmware on my private fleet for months now.

Awesome. Glad it's been of use to you 😊

**Comment by shawaj** [comment]
FYI to anyone who needs it:

https://support.nebra.com/support/solutions/articles/24000078655

**Comment by shawaj** [comment]
@uros76 I found a way to auto-populate all the configuration variables.

Would you mind being a guinea pig again?

If you go to https://hub.balena.io/organizations/nebraltd/fleets/HELIUM-CONTROLLINO and click on fork fleet, when you create the fleet it should automatically populate all the environment variables and config variables for you.

Let me know if it works!

**Comment by uros76** [comment]
> @uros76 I found a way to auto-populate all the configuration variables.
> 
> Would you mind being a guinea pig again?
> 
> If you go to https://hub.balena.io/organizations/nebraltd/fleets/HELIUM-CONTROLLINO and click on fork fleet, when you create the fleet it should automatically populate all the environment variables and config variables for you.
> 
> Let me know if it works!

Seems to be working. Good job! 
Is "HELIUM_MINER_HEIGHT_URL" also needed?

**Comment by shawaj** [comment]
Great thanks @uros76 😊🙏

The Helium miner height URL isn't really needed anymore but we put it in in any case on our main fleets so included in here as well.

**Comment by shawaj** [comment]
Just FYI, I've found an alternative (and better) way to the `gpio-shutdown` dtoverlay mentioned in this comment - https://github.com/NebraLtd/helium-controllino/issues/1#issuecomment-1350774547

> - Still on configuration section, activate "Define DT overlays" and edit this text including quotation signs "spi0-1cs","gpio-shutdown,gpio_pin=17,active_low=1,gpio_pull=up"

The better way is to set:
- "Define DT overlays" and edit this text including quotation signs "spi0-1cs"
- "Allows GPIO pins to be set to specific modes and values at boot time" and edit this text including quotation signs "17=op,dh"

Whilst unlikely, the way using the gpio-shutdown overlay could potentially have unintended consequences (shutdowns) if there is some issue with power supply or something that causes pin 17 to be pulled low externally. With the updated method, this is not a risk.

**Comment by pixelcatcher** [comment]
well... 
who can help? :)

<img width="746" alt="Bildschirmfoto 2023-01-26 um 17 27 42" src="https://user-images.githubusercontent.com/33382133/214892982-42b79262-7e6c-4e66-a56e-1a42e0b637c5.png">

**Comment by shawaj** [comment]
> well... who can help? :)
> 
> <img alt="Bildschirmfoto 2023-01-26 um 17 27 42" width="746" src="https://user-images.githubusercontent.com/33382133/214892982-42b79262-7e6c-4e66-a56e-1a42e0b637c5.png">

Are you getting any error messages when you push the image?

Or on the dashboard there?

**Comment by pixelcatcher** [comment]
honestly I don't know how to push this image. I have already deployed a device but there seems to be no helium related SW on it. Also there is no dashboard. Appartently its because of the error.

**Comment by shawaj** [comment]
Please follow the instructions here https://nebra.io/controllino

**Comment by shawaj** [comment]
Sorry that was the wrong link. This is the correct one https://support.nebra.com/support/solutions/articles/24000078655

**Comment by pixelcatcher** [comment]
Thank you! 
On the unofficial helium discord there is a pinned message with a tutorial which says:

"- Visit https://nebra.io/controllino and click on "Fork this fleet""

Which is what I did.
I will now repeat following your instructions.

**Comment by shawaj** [comment]
Fork this fleet is not what you want to do unless you are fairly technical with experience of docker containers etc.

For auto updates etc you want https://support.nebra.com/support/solutions/articles/24000078655

Maybe you can ask them to update?

**Comment by pixelcatcher** [comment]
I did :) thanks for your help! 
waiting for the device now :) it seems busy

**Comment by shawaj** [comment]
> I did :) thanks for your help! waiting for the device now :) it seems busy

No problem at all. The fork fleet is quite manual and requires you to do your own updates etc

**Comment by acaltas** [comment]
any new Updates for the Controllino Miners with Nebra Software?
@shawaj

**Comment by shawaj** [comment]
What update do you need?

**Comment by acaltas** [comment]
i was only looking what is the latest update for nebra Miners.. and latest was on the 28.Oct 2023.
until now no updates.. befor we got more updates in the months

**Comment by shawaj** [comment]
@acaltas there hasn't been any helium network updates since then
